Documentation/contributors/design-proposals/subscription-status.md

```diff
@@ -0,0 +1,188 @@
+
+    // SubscriptionPackageChannelInvalid indicates the package and channel specified in the SubscriptionSpec is not valid.
+    SubscriptionPackageChannelInvalid SubscriptionConditionType = "PackageChannelInvalid"
+
+    // SubscriptionInstallPlanFailed indicates the InstallPlan responsible for installing the current CSV has failed.
+    SubscriptionInstallPlanFailed SubscriptionConditionType = "InstallPlanFailed"
+
+    // SubscriptionInstallPlanMissing indicates the InstallPlan responsible for installing the current CSV is missing.
+    SubscriptionInstallPlanMissing SubscriptionConditionType = "InstallPlanMissing"
+
+    // SubscriptionInstallPlanAwaitingManualApproval indicates the InstallPlan responsible for installing the current CSV is waiting 
+    // for manual approval.
+    SubscriptionInstallPlanAwaitingManua.lApproval SubscriptionConditionType = "InstallPlanAwaitingManualApproval"
+
+    // SubscriptionInstalledCSVReplacementAvailable indicates there exists a replacement for the installed CSV.
+    SubscriptionInstalledCSVReplacementAvailable SubscriptionConditionType = "InstalledCSVReplacementAvailable"
+
+    // SubscriptionInstalledCSVMissing indicates the installed CSV is missing.
+    SubscriptionInstalledCSVMissing SubscriptionConditionType = "InstalledCSVMissing"
+
+    // SubscriptionInstalledCSVFailed indicates the installed CSV has failed.
+    SubscriptionInstalledCSVFailed SubscriptionConditionType = "InstalledCSVFailed"
+)
+
+type SubscriptionCondition struct {
+    // Type is the type of Subscription condition.
+    Type               SubscriptionConditionType   `json:"type" description:"type of Subscription condition"`
+
+    // Status is the status of the condition, one of True, False, Unknown.
+    Status             corev1.ConditionStatus    `json:"status" description:"status of the condition, one of True, False, Unknown"`
+
+    // Reason is a one-word CamelCase reason for the condition's last transition.
+    // +optional
+    Reason             string            `json:"reason,omitempty" description:"one-word CamelCase reason for the condition's last transition"`
+
+    // Message is a human-readable message indicating details about last transition.
+    // +optional
+    Message            string            `json:"message,omitempty" description:"human-readable message indicating details about last transition"`
+
+    // LastHeartbeatTime is the last time we got an update on a given condition
+    // +optional
+    LastHeartbeatTime  *metav1.Time  `json:"lastHeartbeatTime,omitempty" description:"last time we got an update on a given condition"`
+
+    // LastTransitionTime is the last time the condition transit from one status to another
+    // +optional
+    LastTransitionTime *metav1.Time  `json:"lastTransitionTime,omitempty" description:"last time the condition transit from one status to another"`
+}
+```
+
+### Changes to Subscription Reconciliation
```

Comment:
Is this detailed enough for a proposal, or should there be more specific lines of code pointed out?

Reply:
By "current objects", are you referring to control-loops as well?